up-l1/README.adoc

```diff
@@ -215,11 +188,7 @@ This method implements the _pull_ <<Delivery Method>> for the underlying messagi
 
 [source]
 ----
-receive(
-  IN sourceFilter: UUri,
-  IN sinkFilter: UUri?,
-  OUT result: Future<UMessage&, UStatus>
-)
+receive(sourceFilter: UUri, sinkFilter: UUri [0..1]) : Future<UMessage>
```

Comment:
Here's a thought -- I think for _most_ folks, they will actually want to block here.

Maybe we can have a short snippet like --

> If you wish to treat this function as synchronous, you may use your language of choice's mechanism for doing so. For example, Java's `CompleteableFuture::complete()` or Rust's `.await`.

Dunno. Too hand-holdy?

@stevenhartley -- sanity check me on the Java point :laughing:

Reply:
> Here's a thought -- I think for most folks, they will actually want to block here.

FMPOV the important thing about `receive` is, that it should fail _immediately_ if no message is available, i.e. it should not wait for a message becoming available.

It is not really about whether a particular implementation actually uses async or sync calls. However, IMHO a UTransport implementation should follow a consistent approach across all methods, i.e. either sync or async. If client code needs to be able to deal with async methods anyway, because all of the other methods are async, then I do not see much value in this particular method being the only one defined as sync.

In general, when it comes to IO operations (and UTransport is as close to IO as it gets), using async non-blocking calls has been proven to make systems more performant, scalable and less resource consuming. This also matches my personal experience with other (communication infrastructure) projects ...

Reply:
Please view this as non-blocking for merge. It's a minor point that could be discussed later; more a tip to help a developer, but may just as easily fit in with example code we may write later.